packages/flutter/lib/src/widgets/localizations.dart

```diff
@@ -42,11 +42,13 @@ Future<Map<Type, dynamic>> _loadAll(Locale locale, Iterable<LocalizationsDelegat
   final Map<Type, dynamic> output = <Type, dynamic>{};
   List<_Pending> pendingList;
 
-  // Only load the first delegate for each delgate type.
+  // Only load the first delegate for each delgate type that supports
+  // locale.languageCode.
+  //
   final Set<Type> types = new Set<Type>();
   final List<LocalizationsDelegate<dynamic>> delegates = <LocalizationsDelegate<dynamic>>[];
   for (LocalizationsDelegate<dynamic> delegate in allDelegates) {
-    if (!types.contains(delegate.type)) {
+    if (delegate.isSupported(locale) && !types.contains(delegate.type)) {
```

Comment:
i would flip the order in which you make these calls, so that we don't contact the delegate if we don't need to.